From 35649d2e64a11b2d5c914e24e203c8ca1a595619 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Mon, 24 Jan 2022 17:14:19 +0100
Subject: [PATCH 0638/1695] plat-stm32mp1: conf: add hse monitoring support on
 stm32mp13

Add the possibility to detect HSE over frequency detection
on STM32MP13 devices.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 45b2822634d15acc2a4746e9308d2365bb21be6e plat-stm32mp1: conf: add hse monitoring support on stm32mp13]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: I5934c610fed66615aca48bc884636733e8c6ad0f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/286272
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index aa09b13f0..9a1ceb4c0 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -163,6 +163,7 @@ $(call force,CFG_DRIVERS_CLK_FIXED,y)
 $(call force,CFG_SECONDARY_INIT_CNTFRQ,n)
 $(call force,CFG_STM32_EXTI,y)
 $(call force,CFG_STM32_GPIO,y)
+$(call force,CFG_STM32_HSE_MONITORING,y)
 $(call force,CFG_STM32_VREFBUF,y)
 $(call force,CFG_STM32MP_CLK_CORE,y)
 $(call force,CFG_STM32MP1_OPTEE_IN_SYSRAM,n)
@@ -298,6 +299,10 @@ ifeq ($(call cfg-one-enabled, CFG_STM32_CRYP CFG_STM32_SAES),y)
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
 endif
 
+ifeq ($(CFG_STM32_HSE_MONITORING),y)
+$(call force,CFG_STM32_LPTIMER,y)
+endif
+
 ifeq ($(call cfg-one-enabled, CFG_STM32_LPTIMER CFG_STM32_TIM),y)
 $(call force,CFG_COUNTER_DRIVER,y)
 endif
-- 
2.47.2

